### PR TITLE
[CI/CD] iOS and tvOS build and distribution

### PR DESCRIPTION
**Pull request by ldanilina**
- using fastlane to build and distribute iOS / tvOS app

### CODE REVIEW

File: .github/workflows/cd_ios.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.1.2
+
+      - name: Setup Fastlane
+        run: |
+          cd ios
+          bundle install
+          bundle exec pod install
```
Comment (aravind-raveendran) [inline .github/workflows/cd_ios.yml:35]: `gem install bundler` and `bundle install` can be moved here, so we don't need it in the step below?

File: ios/Gemfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+source "https://rubygems.org"
+
```
Comment (aravind-raveendran) [inline ios/Gemfile:2]: Now that we removed Gemfile in the root, can we include the below in this file ?

ruby '3.1.2'

gem 'cocoapods', '1.14.3'

File: ios/fastlane/.env.default [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// .env.default
+XCODE_SELECT_PATH = /Applications/Xcode_15.0.1.app
```
Comment (aravind-raveendran) [inline ios/fastlane/.env.default:2]: May be use `Xcode_15.1.app` as its the latest available in GitHub actions runners?

File: ios/Gemfile (hunk 2 above)
Comment (ldanilina) [reply ios/Gemfile:7]: Thanks a lot!

**Review by RameshDPrasad — COMMENTED** [review]
Do we need a separate YAML for TVOS?

**Comment by ldanilina** [comment]
> Do we need a separate YAML for TVOS?

Combined the building iOS and tvOS jobs into a single file.

@aravind-raveendran do you have a better idea of organising the jobs? You've mentioned to run them in parallel, does that mean we should repeat the setup twice?

**Comment by aravind-raveendran** [comment]
> > Do we need a separate YAML for TVOS?
> 
> Combined the building iOS and tvOS jobs into a single file.
> 
> @aravind-raveendran do you have a better idea of organising the jobs? You've mentioned to run them in parallel, does that mean we should repeat the setup twice?

Hi @ldanilina - Yes, I had two separate jobs in mind? Do you know if two jobs normally are allocated to the same runner ?

**Comment by ldanilina** [comment]
> Hi @ldanilina - Yes, I had two separate jobs in mind? Do you know if two jobs normally are allocated to the same runner ?

For each job there's a separate runner (and one runner can only run one job at a time).

**Comment by aravind-raveendran** [comment]
> > Hi @ldanilina - Yes, I had two separate jobs in mind? Do you know if two jobs normally are allocated to the same runner ?
> 
> For each job there's a separate runner (and one runner can only run one job at a time). So if we need to run them in parallel, we should repeat the setup steps for each job..



> > Do we need a separate YAML for TVOS?
> 
> Combined the building iOS and tvOS jobs into a single file.
> 
> @aravind-raveendran do you have a better idea of organising the jobs? You've mentioned to run them in parallel, does that mean we should repeat the setup twice?

I have not tested this, you current solution looks good to me.

Would the below help parallelise the builds?

name: Build for iOS and tvOS

jobs:
  setup(clone, lfs, bundler):
   ...

  build_for_ios:
    needs: setup
    ...

  build_for_tvos:
    needs: setup
    ...

**Review by aravind-raveendran — APPROVED** [review]
LGTM

**Comment by ldanilina** [comment]
> I have not tested this, would the below help parallelise the builds?
> 
> ```
> name: Build for iOS and tvOS
> 
> jobs:
>   setup(clone, lfs, bundler):
>    ...
> 
>   build_for_ios:
>     needs: setup
>     ...
> 
>   build_for_tvos:
>     needs: setup
>     ...
> ```

The problem here is that build_for_ios and build_for_tvos will run in their own clean environment and won't have access to the setup.
We probably can make the setup as a separate action and reuse it in each job.

**Pull request merged**